packages/core/contracts/polygon/GovernorChildTunnel.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "../external/polygon/tunnel/FxBaseChildTunnel.sol";
+
+/**
+ * @title Governor contract deployed on sidechain that receives governance actions from Ethereum.
+ */
+contract GovernorChildTunnel is FxBaseChildTunnel {
+    event ExecutedGovernanceTransaction(address indexed to, uint256 value, bytes data);
+
+    constructor(address _fxChild) FxBaseChildTunnel(_fxChild) {}
+
+    /**
+     * @notice Executes governance transaction created on Ethereum.
+     * @dev The data will be received automatically from the state receiver when the state is synced between Ethereum
+     * and Polygon. This will revert if the Root chain sender is not the `fxRootTunnel` contract.
+     * @param sender The sender of `data` from the Root chain.
+     * @param data ABI encoded params with which to call `_publishPrice`.
+     */
+    function _processMessageFromRoot(
+        uint256, /* stateId */
+        address sender,
+        bytes memory data
+    ) internal override validateSender(sender) {
+        // TODO: Where do we get `value` amount of MATIC from if its >0? The sender of this method will be the system
```

Comment:
@mrice32 did you consider this when implementing the Source/Sink Governor? Similar logic implemented in `Governor.sol` [here](https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/oracle/implementation/Governor.sol#L153)